community/detectors/jira_cve_2022_0540/src/main/java/com/google/tsunami/plugins/detectors/jira/Cve20220540VulnDetector.java

```diff
@@ -0,0 +1,174 @@
+    } else {
+      // Assume the service uses HTTP protocol when the scanner cannot identify the actual service.
+      targetUrlBuilder
+          .append("http://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    targetUrlBuilder.append(url);
+    return targetUrlBuilder.toString();
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("Cve20220540VulnDetector starts detecting.");
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(Cve20220540VulnDetector::isWebServiceOrUnknownService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String insightUrl = buildTargetUrl(networkService, INSIGHT_CHECK_VUL_PATH);
+    try {
+      HttpResponse httpResponse =
+          httpClient.send(get(insightUrl).withEmptyHeaders().build(), networkService);
+      if (httpResponse.status().code() == 200
+          && httpResponse.bodyString().get().contains(INSIGHT_BODY)) {
+        return true;
+      }
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", networkService);
+    }
+
+    String wbsGanttUrl = buildTargetUrl(networkService, WBSGANTT_CHECK_VUL_PATH);
+    try {
+      HttpResponse httpResponse =
+          httpClient.send(get(wbsGanttUrl).withEmptyHeaders().build(), networkService);
+      if (httpResponse.status().code() == 200
+          && httpResponse.bodyString().get().contains(WBSGANTT_DOBY)) {
+        return true;
+      }
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", networkService);
+    }
+
+    return false;
+  }
+
+  private DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder()
+                        .setPublisher("TSUNAMI_COMMUNITY")
+                        .setValue("CVE_2022_0540"))
+                .setSeverity(Severity.CRITICAL)
+                .setTitle(
+                    "CVE-2022-0540: unauthenticated attacker to bypass authentication by sending a"
+                        + " specially crafted HTTP request")
```

Comment:
Suggested title: `CVE-2022-0540: Authentication Bypass in Atlassian Jira Service Management Server and Data Center`